backend/prompt_studio/prompt_studio/views.py

```diff
@@ -41,3 +47,115 @@ def get_queryset(self) -> Optional[QuerySet]:
+        """
+        try:
+            # Validate request data
+            serializer = ReorderPromptsSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+
+            # Extract validated data from the serializer
+            start_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.START_SEQUENCE_NUMBER
+            )
+            drop_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.DROP_SEQUENCE_NUMBER
+            )
+            prompt_id = serializer.validated_data.get(ToolStudioPromptKeys.PROMPT_ID)
+
+            prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.get(
+                pk=prompt_id
+            )
+            filtered_prompts_data = []
+            tool_id = prompt_instance.tool_id
+
+            # Determine the direction of sequence adjustment based on start and
+            # drop sequence numbers
+            if start_sequence_number < drop_sequence_number:
+                logger.info(
+                    "Start sequence number is less than drop sequence number. "
+                    "Decrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__gt": start_sequence_number,
+                    "sequence_number__lte": drop_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=False
+                )
+
+            elif start_sequence_number > drop_sequence_number:
+                logger.info(
+                    "Start sequence number is greater than drop sequence number. "
+                    "Incrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__lt": start_sequence_number,
+                    "sequence_number__gte": drop_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=True
+                )
+
+            # If the start and drop sequence numbers are equal,
+            # return a bad request response
+            else:
+                logger.warning(
+                    "Start and drop sequence numbers are equal. "
+                    "Returning bad request response."
+                )
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={"detail": "Start and drop sequence numbers are equal."},
+                )
```

Comment:
@tahierhussain add this in the `ReorderPromptsSerializer` as a validation error. You can refer [this](https://github.com/Zipstack/unstract/blob/5c2a4e9a747d88358993321d51badffc83b4686a/backend/tenant_account/serializer.py#L55-L71) or [this](https://github.com/Zipstack/unstract/blob/e2ed9089525fddf25fd186743f1fd7b4de15a9e0/backend/api/serializers.py#L63-L68)

Reply:
@chandrasekharan-zipstack Thanks for the suggestion. Done.